### PR TITLE
Added validation set loss tracking

### PR DESCRIPTION
**Pull request by larkz**
- Add validation set loss tracking for early stopping.

### CODE REVIEW

**Comment by kashif** [comment]
Looks good

File: pts/trainer.py [hunk 1]
```diff
@@ -62,27 +61,59 @@ def __call__(
...
-                for batch_no, data_entry in enumerate(it, start=1):
+                for batch_no, data_entry in train_iter_obj:
+
                     optimizer.zero_grad()
-                    inputs = [v.to(self.device) for v in data_entry.values()]
 
+                    # Strong assumption that validation_iter and train_iter are same iter size
+                    if validation_iter is not None:
+                        with torch.no_grad():
+                            val_data_entry = val_iter_obj[batch_no-1][1]
```
[the remark this answers is not in the document]
Comment (larkz) [reply pts/trainer.py:80]: So `val_iter_obj` is a zip of the index of `validation_iter` and `validation_iter` itself,  by accessing the second element we get the validation data, this allows us to access the corresponding validation_iter, via the  train_iter index.
